payments-contracts/src/main/kotlin/com/gevamu/corda/contracts/PaymentContract.kt

```diff
@@ -16,24 +16,241 @@
+        override val spec: VerifierSpec
+    ) : PaymentContractVerifier {
+        override fun verify() {
+            if (spec.output == null) {
+                throw IllegalArgumentException("The transaction is expected to have an output")
+            }
+            if (!spec.command.signers.contains(spec.output.gateway.owningKey)) {
+                throw IllegalArgumentException("The transaction is not signed by the Gateway (${spec.output.gateway}), output index ${spec.outputIndex}")
+            }
+        }
+    }
+
+    class PayerSignerVerifier(
+        override val spec: VerifierSpec
+    ) : PaymentContractVerifier {
+        override fun verify() {
+            if (spec.output == null) {
+                throw IllegalArgumentException("The transaction is expected to have an output")
+            }
+            if (!spec.command.signers.contains(spec.output.payer.owningKey)) {
+                throw IllegalArgumentException("The transaction is not signed by the Payer (${spec.output.payer})")
+            }
+        }
+    }
+
+    class RequiredFieldsVerifier(
+        override val spec: VerifierSpec
+    ) : PaymentContractVerifier {
+        override fun verify() {
+            if (spec.output == null) {
+                throw IllegalArgumentException("The transaction is expected to have an output")
+            }
+            if (spec.output.endToEndId.isBlank()) {
+                throw IllegalArgumentException("The field EndToEndId is blank for unique payment id ${spec.output.uniquePaymentId}, output index ${spec.outputIndex}")
+            }
+        }
+    }
+
+    class PaymentConsensusVerifier(
+        override val spec: VerifierSpec
+    ) : PaymentContractVerifier {
+        override fun verify() {
+            if (spec.input == null) {
+                throw IllegalArgumentException("The transaction is expected to have an input")
+            }
+            if (spec.output == null) {
+                throw IllegalArgumentException("The transaction is expected to have an output")
+            }
+            if (spec.input.endToEndId != spec.output.endToEndId) {
+                throw IllegalArgumentException("EndToEndId changed for unique payment id ${spec.input.uniquePaymentId}, output index ${spec.outputIndex}")
+            }
+            if (spec.input.payer != spec.output.payer) {
+                throw IllegalArgumentException("Payer changed for unique payment id ${spec.input.uniquePaymentId}, output index ${spec.outputIndex}")
+            }
+            if (spec.input.gateway != spec.output.gateway) {
+                throw IllegalArgumentException("Gateway changed for unique payment id ${spec.input.uniquePaymentId}, output index ${spec.outputIndex}")
+            }
+        }
+    }
+
+    class PaymentStatusTransitionVerifier(
+        override val spec: VerifierSpec
+    ) : PaymentContractVerifier {
+        override fun verify() {
+            val isValid = when (spec.command.value) {
```

Comment:
You use this verifier from verifier classes for specific command. Checking command again looks strange